model/release.go

```diff
@@ -146,11 +146,49 @@ func (tag *Tag) UnmarshalJSON(b []byte) error {
 	return nil
 }
 
+type Notes string
+
+var (
+	NotesLengthMaximumCharacters = 1024
+
+	ErrReleaseNotesTooLong  = errors.New("release notes too long")
+	ErrCharactersNotAllowed = errors.New("release notes contain characters which are not allowed")
+)
+
+func allowedByte(c byte) bool {
+	// at this point I am unsure if we can restrict the charset for the notes, it will make
+	// it difficult to use
+	return true
+}
+
+func (n Notes) Validate() error {
+	length := len(n)
+	if length > NotesLengthMaximumCharacters {
+		return ErrReleaseNotesTooLong
+	}
+	for i := 0; i < length; i++ {
+		if !allowedByte(n[i]) {
+			return ErrCharactersNotAllowed
+		}
+	}
```

Comment:
You could use `unicode.IsGraphic` to check that the text is printable. As for XSS, the json library escapes HTML by default when marshalling. You could alternatively make `json.Unmarshaler` that calls `json.HTMLEscape` on the decoded string if you're paranoid of storing strings without escaping them first :sweat_smile: 
```go
// Error type is optional, but it helps the end-user to understand what is not allowed.
type InvalidCharError struct {
	Offset int
	Char   rune
}

func (err *CharError) Error() string {
	return fmt.Sprintf(`invalid character '%c' at offset %d`, err.Char, err.Offset)
} 
```
```suggestion
	if i := strings.IndexFunc(string(n), unicode.IsPrint); i > 0 {
		return InvalidCharError{
			Char: n[i],
			Offset: i,
		}
	}
```